src/modules/hip/kernel/resize.hpp

```diff
@@ -0,0 +1,735 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resize device helpers --------------------
+
+__device__ void resize_roi_and_srclocs_hip_compute(int4 *srcRoiPtr_i4, uint2 *dstDimsWH, int id_x, int id_y, d_float16 *locSrc_f16)
+{
+    float wRatio = (float)(srcRoiPtr_i4->z - srcRoiPtr_i4->x + 1) / dstDimsWH->x;
+    float hRatio = (float)(srcRoiPtr_i4->w - srcRoiPtr_i4->y + 1) / dstDimsWH->y;
+    float4 wOffset_f4 = (float4)((wRatio - 1) * 0.5f);
+    float4 hOffset_f4 = (float4)((hRatio - 1) * 0.5f);
+
+    d_float8 increment_f8, locDst_f8x, locDst_f8y;
+    increment_f8.f4[0] = make_float4(0.0f, 1.0f, 2.0f, 3.0f);
+    increment_f8.f4[1] = make_float4(4.0f, 5.0f, 6.0f, 7.0f);
+    locDst_f8x.f4[0] = (float4)id_x + increment_f8.f4[0];
+    locDst_f8x.f4[1] = (float4)id_x + increment_f8.f4[1];
+    locDst_f8y.f4[0] = (float4)id_y;
+    locDst_f8y.f4[1] = (float4)id_y;
+
+    locSrc_f16->f8[0].f4[0] = (locDst_f8x.f4[0] * (float4)wRatio) + wOffset_f4 + (float4)srcRoiPtr_i4->x;  // Compute src x locations in float for dst x locations [0-3]
+    locSrc_f16->f8[0].f4[1] = (locDst_f8x.f4[1] * (float4)wRatio) + wOffset_f4 + (float4)srcRoiPtr_i4->x;  // Compute src x locations in float for dst x locations [4-7]
+    locSrc_f16->f8[1].f4[0] = (locDst_f8y.f4[0] * (float4)hRatio) + hOffset_f4 + (float4)srcRoiPtr_i4->y;  // Compute src y locations in float for dst y locations [0-3]
+    locSrc_f16->f8[1].f4[1] = (locDst_f8y.f4[1] * (float4)hRatio) + hOffset_f4 + (float4)srcRoiPtr_i4->y;  // Compute src y locations in float for dst y locations [4-7]
+}
+
+__device__ void resize_roi_generic_srcloc_and_weight_hip_compute(int roiLoc, int dstLocation, float scale, int limit, int *srcLoc, float *weight, float offset, int srcStride)
+{
+    float srcLocationFloat = ((float) dstLocation) * scale + offset + (float)roiLoc;
+    int srcLocation = (int)ceilf(srcLocationFloat);
+    *weight = srcLocation - srcLocationFloat;
```

Comment:
You can combine 30 and 31 into 1 statement (*weight = ceilf(srcLocationFloat) - srcLocationFloat;)

Reply:
We are using the rounded value (srcLocation) from Line 30 in [Line 32](https://github.com/r-abishek/rpp/blob/ar/opt_resize_hip/src/modules/hip/kernel/resize.hpp#L32) 
*srcLoc = ((srcLocation > limit) ? limit : srcLocation) * srcStride;

if we combine Line 30 and Line 31
we need to call ceilf 2 additional times in Line 32 as below which can be costly
*srcLoc = (((int)ceilf(srcLocation) > limit) ? limit : ceilf(srcLocation)) * srcStride;